### PR TITLE
Update ORM requirements to allow DBAL 2.5 again

### PR DESCRIPTION
**Pull request by Tobion**
@Ocramius could doctrine release ORM 2.4.8 so https://github.com/doctrine/doctrine2/commit/e05930e71482df06982c332b8820a1ae51bd7f28 is tagged? I think orm <2.4.8 is incompatible with DBAL 2.5.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -9,8 +9,7 @@
     "require": {
         "php": ">=5.3.3",
         "symfony/symfony": "2.3.*",
-        "doctrine/orm": "~2.2,>=2.2.3,<2.5",
-        "doctrine/dbal": "<2.5",
+        "doctrine/orm": "~2.2.3||~2.3.0||~2.4.8||~2.5",
```
Comment (nicolas-grekas) [inline composer.json]: single `|`?

Comment (Tobion) [inline composer.json]: is the doc wrong https://getcomposer.org/doc/01-basic-usage.md#package-versions ?

Comment (stof) [inline composer.json]: bith are valid as of alpha10. A single ``|`` works on older versions

Comment (stof) [inline composer.json]: And I suggest dropping 2.2 and 2.3 here. They are EOLed versions of Doctrine

Comment (Tobion) [inline composer.json]: changed. I also used the caret operator now. I guess it's time. And it's the responsibility of the people manually updating their dependencies anyway.

File: composer.json [hunk 2]
```diff
@@ -9,15 +9,14 @@
     "require": {
         "php": ">=5.3.3",
         "symfony/symfony": "2.3.*",
-        "doctrine/orm": "~2.2,>=2.2.3,<2.5",
-        "doctrine/dbal": "<2.5",
+        "doctrine/orm": "^2.4.8",
         "doctrine/doctrine-bundle": "~1.2",
         "twig/extensions": "1.0.*",
```
Comment (stof) [inline composer.json]: this one should be changed to ``~1.0`` to allow to use uptodate versions (the latest stable is 1.2.0)

**Comment by Ocramius** [comment]
@Tobion ORM 2.4.7 works with DBAL 2.5.x

**Comment by Tobion** [comment]
@Ocramius what do you mean with "works"? Does it not cause #748? How did you fix it for these version?

**Comment by Ocramius** [comment]
It's not fixed in ORM 2.5 either.

Marco Pivetta

http://twitter.com/Ocramius

http://ocramius.github.com/

**Comment by Tobion** [comment]
Then why does the error not appear anymore in ORM 2.5 if it's not fixed?

**Comment by Ocramius** [comment]
Luck? I see no explicit fix happening.

That said, ORM 2.5 is DBAL 2.5 compatible, just platform auto-detection in DBAL is broken.

**Comment by stof** [comment]
@Ocramius the auto-detection implementation in DBAL is fine: it is delayed until you retrieve the Platform from the connection (at which point the result of the detection is needed). As I explained several times already (including in http://www.doctrine-project.org/jira/browse/DBAL-1057 which tracks this officially), the issue is that the ORM was retrieving the platform in many places in the constructor instead of doing it only when it actually needs it. This means that just instantiating the ORM was asking for the platform at the right version.
The commit referenced above fixed this issue in 1 place in the ORM, and other places getting the platform in their constructor are the persisters, the hydrators and the DQL function parsers. None of them is instantiated until they are used IIRC, meaning that the requirement to be connected seems legitimate.
So I think it might be fixed by the commit referenced above.

**Comment by Tobion** [comment]
Not yet. Doctrine needs to tag a new release which seems very hard to do. See https://github.com/doctrine/DoctrineBundle/issues/351#issuecomment-98545563

**Comment by Ocramius** [comment]
s/hard/time-consuming-during-my-evening-that-is-already-fully-booked-for-these-weeks
:-)

Marco Pivetta

http://twitter.com/Ocramius

http://ocramius.github.com/

**Comment by dbu** [comment]
ping @beberlei can you tag doctrine orm 2.4.8 maybe?

**Comment by stof** [comment]
and also doctrine orm 2.5.1

**Comment by Tobion** [comment]
Doctrine 2.4.8 has been released. So this can be merged.

**Comment by fabpot** [comment]
Thank you @Tobion.

**Comment by derrabus** [comment]
Don't forget to update the lock file. It's now out of sync and references DBAL 2.4.4 which is incompatible with the current requirements from composer.json.

**Comment by GrahamCampbell** [comment]
The lock file gets updated before each release.

**Comment by GrahamCampbell** [comment]
See https://github.com/symfony/symfony-standard/commit/a522d9de5d5d0eb10b4121d9c1d18f35c2ddbe46 for example.

**Comment by GrahamCampbell** [comment]
It's not out of sync.

**Comment by GrahamCampbell** [comment]
Just the newer branches not updated yet since there's not been a release there since.

**Comment by derrabus** [comment]
@GrahamCampbell: I see, thanks for clarifying.

**Pull request merged**